### DISCUSSION TITLE
Table

### DISCUSSION THREAD

**Discussion by boxadesign** [ONSdigital/design-system#102] [Feedback]
Use this place to discuss the [table component](https://ons-design-system.netlify.com/components/table) in the [ONS Design System](https://ons-design-system.netlify.com)

**Comment by jrbarnes9** [comment]
Incorrect param name `trHighlight` in tables component macro-options

Should be just `highlight`

**Reply by jrbarnes9** [reply]
Fixed in #1613

**Comment by jrbarnes9** [comment]
Sortable table header row buttons mis-aligned, missing `role` and incorrectly styled

Header row buttons should be:
- inheriting link style `text-underline-position` 
- accessible with `role=link` attribute
- left-aligned with column cells content

![Screenshot 2021-10-05 at 16 02 38](https://user-images.githubusercontent.com/43346934/136050434-8c5a567c-a8cb-47ae-aada-7f786ac01e60.png)

**Reply by jrbarnes9** [reply]
Raised #1723

**Comment by jrbarnes9** [comment]
Table variants should use `variants` parameter instead of classes

**Reply by jrbarnes9** [reply]
Raised #1725

**Comment by rpcatling** [comment]
I believe there is a problem with `thClasses` macro option.
When I set the value, eg:
```
thClasses='my_class'
```
Then the rendered HTML has this:
```
<th scope="col" class="ons-table__headermy_class" aria-sort="descending">
```

ie the class provided is directly concatenated with `ons-table__header` rather than having a space between them, which causes the table header to render badly.

Until the bug is fixed, the workaround is simply to provide the space when creating the value:
```
thClasses=' my_class'
```

**Reply by jrbarnes9** [reply]
Raised #1812

**Comment by jrbarnes9** [comment]
Sortable table requires all columns to be sortable

**Reply by jrbarnes9** [reply]
Raised #2125

**Comment by jrbarnes9** [comment]
Enable table header classes to visually hide sortable table header cell contents

Raised in DS forum

**Reply by jrbarnes9** [reply]
#2332

**Comment by jrbarnes9** [comment]
Only add `tabindex` to `ons-table-scrollable__content` if the table doesn't fit in the viewport. When it doesn't need to be scrolled, the user shouldn't be able to give focus to the table content.

![Screenshot 2022-07-25 at 16 29 36](https://user-images.githubusercontent.com/43346934/180816926-24fe30e2-2077-4355-8ff4-501cd7512ec9.png)

**Reply by jrbarnes9** [reply]
#2354

**Reply by AtinDESNEZ** [reply]
I am working on one table where I have more than 4 columns (most accessible) but not sure if that would be appreciated from accessibility perspective cos in  a mobile view only 4 columns are visible without horizontal scroll. I noticed you have 7-8 columns in your example above. If you could please help me understand how does this work from accessibility point of view.

**Comment by jrbarnes9** [comment]
Sortable table headers with spaces create invalid `id`

![Screenshot 2022-08-12 at 14 23 56](https://user-images.githubusercontent.com/43346934/184362971-1175f3ed-d170-4435-816a-3e1f2e1247d6.png)

**Reply by jrbarnes9** [reply]
Raised #2397

**Comment by jhmjam** [comment]
Hi, I noticed when using two scrollable tables on the same html page,  if the header values for the tables are different then there is no issue likewise if i remove the sortable variant it is ok but with the variant and both tables sharing the same header values we get this bunch up where the second tables header values are concatenated with the first tables header values.

Can you advise how i get the headers into the correct place please?

Thanks

**Reply by jrbarnes9** [reply]
Hi @jhmjam I've raised a bug for this: #2601

**Reply by jhmjam** [reply]
@jrbarnes9 Thanks :)
